core/src/main/java/com/linecorp/armeria/client/AggregatedResponseAs.java

```diff
@@ -57,10 +57,6 @@ static <T> ResponseAs<AggregatedHttpResponse, ResponseEntity<T>> json(TypeRefere
 
     private static <T> ResponseEntity<T> newJsonResponseEntity(AggregatedHttpResponse response,
                                                                JsonDecoder<T> decoder) {
-        if (!response.status().isSuccess()) {
```

Comment:
If we don't throw an exception when a non-2xx status is received, users may see `JsonProcessingException` during deserialization. As `JsonProcessingException` does not have response information, users find it difficult to know what response was received from the exception.

It would be better to raise an `InvalidHttpResponseException` for non-2xx status by default.

Reply:
Thanks for the input 😄 

Then let's go with this option:
> 1. Assume that [BlockingWebClientRequestPreparation|RestClientPreparation]#asJson has an implicit predicate where response.status().isSuccess() == true.
> 2. Modify newJsonResponseEntity to accept a predicate, and throw an exception if the predicate is false